src/main/java/org/openhab/tools/analysis/checkstyle/ImportExportedPackagesCheck.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.ivy.osgi.core.BundleRequirement;
+import org.apache.ivy.osgi.core.ExportPackage;
+import org.apache.ivy.osgi.core.ManifestParser;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+
+/**
+ * Checks if all of the exported packages are imported by the bundle itself
+ *
+ * @author Mihaela Memova
+ */
+public class ImportExportedPackagesCheck extends AbstractStaticCheck {
+
+    public static final String NOT_IMPORTED_PACKAGE_MESSAGE = "The exported package is not imported";
+    public static final String MANIFEST_EXTENSTION = "MF";
+    public static final String EXPORT_PACKAGES_HEADER = "Export-Package:";
+
+    Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public ImportExportedPackagesCheck() {
+        setFileExtensions(MANIFEST_EXTENSTION);
+    }
+
+    @Override
+    protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
+        int exportsStartIndex = 0;
+        for (int i = 0; i < lines.size(); i++) {
+            if (lines.get(i).startsWith(EXPORT_PACKAGES_HEADER)) {
+                exportsStartIndex = i;
+            }
+        }
+
+        try {
+            Set<ExportPackage> exports = ManifestParser.parseManifest(file).getExports();
+            Set<BundleRequirement> imports = ManifestParser.parseManifest(file).getImports();
+
+            for (Object export : exports) {
+                if (!isPackageImported(imports, export)) {
+                    int lineToLog = findLineNumber(lines, export.toString(), exportsStartIndex);
+                    log(lineToLog, NOT_IMPORTED_PACKAGE_MESSAGE);
+                }
+            }
+        } catch (IOException e) {
+            logger.error("An error occured while processing the file", e, file.getPath());
```

Comment:
You use no placeholders and the `e` should be final argument, also apply to line 65